src/main/java/com/firebolt/jdbc/client/account/FireboltAccountRetriever.java

```diff
@@ -16,6 +16,7 @@ public class FireboltAccountRetriever<T> extends FireboltClient {
     private final String path;
     private final Class<T> type;
 
+    @SuppressWarnings("java:S107") //Number of parameters (8) > max (7). This is the price of the immutability
     public FireboltAccountRetriever(OkHttpClient httpClient, ObjectMapper objectMapper, FireboltConnection connection, String customDrivers, String customClients, String host, String path, Class<T> type) {
```

Comment:
Just curious, why wouldn't we just make this class with two methods: getId and getEngine?
I see we instantiate it twice with different paths and return types, couldn't those just be two methods of this class?

Reply:
It would then comply with how we implemented FireboltAccountClient for fb1.0

Reply:
I think that my design is more object oriented and thefore more flexible. For example if account has yet another field, e.g. owner I do not have to implement yet another method; just create another instance of this class in place where I need it.

Reply:
I believe that this design better follows the OOD paradigm.